implementations/rust/ockam/ockam_api/src/cli_state/traits.rs

```diff
@@ -196,3 +228,69 @@ pub trait StateItemTrait: Sized {
     fn path(&self) -> &PathBuf;
     fn config(&self) -> &Self::Config;
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::cli_state::{StateDirTrait, StateItemTrait};
+    use std::path::PathBuf;
+
+    #[test]
+    fn test_is_item_path() {
+        let config = TestConfig::new("dir".into());
+        let path = config.path("name");
+        assert!(config.is_item_path(&path).unwrap())
+    }
+
+    /// Dummy configuration
+    struct TestConfig {
+        dir: PathBuf,
+    }
+    impl StateDirTrait for TestConfig {
+        type Item = TestConfigItem;
+        const DEFAULT_FILENAME: &'static str = "";
+        const DIR_NAME: &'static str = "";
+        const HAS_DATA_DIR: bool = false;
+
+        fn new(dir: PathBuf) -> Self {
+            Self { dir }
+        }
+
+        fn dir(&self) -> &PathBuf {
+            &self.dir
+        }
+    }
+
+    struct TestConfigItem {
+        path: PathBuf,
+        config: String,
+    }
+    impl StateItemTrait for TestConfigItem {
+        type Config = String;
+
+        fn new(path: PathBuf, config: Self::Config) -> crate::cli_state::Result<Self> {
+            Ok(TestConfigItem { path, config })
+        }
+
+        fn load(path: PathBuf) -> crate::cli_state::Result<Self> {
+            Ok(TestConfigItem {
+                path,
+                config: "config".into(),
+            })
+        }
+
+        fn path(&self) -> &PathBuf {
+            &self.path
+        }
+
+        fn config(&self) -> &Self::Config {
+            &self.config
+        }
+    }
+}
+
+trait FileSystem {
```

Comment:
Unused?

Reply:
Good catch. This was the remains of an attempt to unit test without calling the file system. I removed it now